packages/xo-server/src/xapi/index.mjs

```diff
@@ -495,17 +495,37 @@ export default class Xapi extends XapiBase {
       bypassAssert = false,
     }
   ) {
+    const srRef = sr !== undefined ? hostXapi.getObject(sr).$ref : undefined
     const getDefaultSrRef = once(() => {
-      if (sr !== undefined) {
-        return hostXapi.getObject(sr).$ref
-      }
       const defaultSr = host.$pool.$default_SR
       if (defaultSr === undefined) {
         throw new Error(`This operation requires a default SR to be set on the pool ${host.$pool.name_label}`)
       }
       return defaultSr.$ref
     })
 
+    // VDIs/SRs mapping
+    // For VDI:
+    // - If a map of VDI -> SR was explicitly passed: use it
+    // - Else if SR was explicitly passed: use it
+    // - Else if VDI SR is reachable from the destination host: use it
+    // - Else: use the migration main SR or the pool's default SR (error if none of them is defined)
+    function getMigrationSrRef(vdi) {
+      if (mapVdisSrs[vdi.$id] !== undefined) {
```

Comment:
As I understand the front, it can't be empty for any disk we want to migrate  , since 
```js
    // 2021-02-16: Fill the map (VDI -> SR) with *all* the VDIs to avoid unexpectedly migrating them to the wrong SRs:
```

Reply:
I need to remove this comment. In reality, the VDI - SR mapping can be empty.
- From the VM view, if we migrate without explicitly assigning an SR to each VDI, then it will be empty, and we will use the SR passed as a parameter as the destination SR if one is given; otherwise, we will use the current VDI SR if accessible, else the default pool SR.
- From the Home view, if we migrate from one shared SR to local storage, it will also be empty and then we will use the SR passed as parameter or the current VDI SR, or the default pool SR.

From what I understand, this "bugfix" was made because sometimes VDIs were migrated to the default SR. See [PR#5578](https://github.com/vatesfr/xen-orchestra/pull/5578) and [gitlab#3355](https://gitlab.com/vates/xoa-support/-/issues/3355).

In the front-end, in the migrate-vms-modal, we previously tried to add the default SR to the migration, even if we didn't want to. I fixed this part to only add the default SR if VDIs need to be migrated.